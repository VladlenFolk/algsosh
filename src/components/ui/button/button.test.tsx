import { Button } from "./button";
import {render, screen, fireEvent} from "@testing-library/react";
import renderer from "react-test-renderer";


describe("Отрисовка кнопки", () => {
  it("с текстом", () => {
    const buttonSnapshot = renderer.create(<Button text="Добавить" />).toJSON();
    expect(buttonSnapshot).toMatchSnapshot();
  });
  it("без текста", () => {
    const buttonSnapshot = renderer.create(<Button />).toJSON();
    expect(buttonSnapshot).toMatchSnapshot();
  });
  it("с блокировкой", () => {
    const buttonSnapshot = renderer.create(<Button disabled={true} />).toJSON();
    expect(buttonSnapshot).toMatchSnapshot();
  });
  it("с загрузкой", () => {
    const buttonSnapshot = renderer.create(<Button isLoader={true} />).toJSON();
    expect(buttonSnapshot).toMatchSnapshot();
  });
  it("вызов коллбека при клике на кнопку",()=>{
    const onClick = jest.fn();
    render (<Button onClick={onClick}>Button</Button>);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(onClick).toHaveBeenCalledTimes(1)
  })
});