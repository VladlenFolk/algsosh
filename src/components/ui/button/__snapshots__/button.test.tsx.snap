// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Отрисовка кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Отрисовка кнопки с блокировкой 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Отрисовка кнопки с загрузкой 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Отрисовка кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Добавить
  </p>
</button>
`;
